src/common/fetchClient.ts

```diff
@@ -0,0 +1,95 @@
+export interface FetchClientConfig {
+  baseURL: string;
+  headers: Record<string, string>;
```

Comment:
Small thing/feel free to ignore but wondering if headers must be required for the client config, though we do pass one in where we use it.

Reply:
Yeah, technically neither are required so I'll make them optional.